File: src/main/java/leetcode/Problem1266.java
package leetcode;

@Difficulty(Level.EASY)
@Algorithms(Algorithm.MATH)
@BeatsPercent(100)
@TimeComplexity(worst = Complexity.LINEAR_N)
@SpaceComplexity(worst = Complexity.CONSTANT)
public class Problem1266 {
    public int minTimeToVisitAllPoints(int[][] points) {
        int time = 0;
        for (int i = 1; i < points.length; ++i) {
            time += Math.max(Math.abs(points[i][0] - points[i - 1][0]), Math.abs(points[i][1] - points[i - 1][1]));
        }
        return time;
    }
}

File: src/problem_1266.py
class Solution:
    def minTimeToVisitAllPoints(self, p: List[List[int]]) -> int:
        res = 0
        x0, y0 = p[0]
        for x, y in p[1:]:
            res += max(abs(x - x0), abs(y - y0))
            x0, y0 = x, y
        return res
